packages/google_maps_flutter/lib/src/cap.dart

```diff
@@ -0,0 +1,49 @@
+part of google_maps_flutter;
+
+/// Immutable cap that can be applied at the start or end vertex of a Polyline.
+class Cap {
```

Comment:
annotate this class with `@immutable`. That way you don't need to mention the immutable nature as a comment and dart analyzer will lint useful stuff, for example [this](https://dart-lang.github.io/linter/lints/prefer_const_constructors_in_immutables.html)